src/contract-ui/tabs/edit-extensions/components/ExtensionForm.tsx

```diff
@@ -0,0 +1,406 @@
+                _dark={{
+                  color: "yellow.500",
+                }}
+                _light={{
+                  color: "yellow.700",
+                }}
+              >
+                Extension may not be compatible
+              </Text>
+            </Flex>
+          )}
+        </FormControl>
+
+        <FormControl isInvalid={!!errors.version} isRequired={true}>
+          <FormLabel>Extension Version</FormLabel>
+          <Skeleton isLoaded={!allVersions.isFetching} borderRadius="lg">
+            <Select
+              disabled={
+                !allVersions.data ||
+                allVersions.isLoading ||
+                isExtensionCompatibleQuery.data === false ||
+                installMutation.isLoading ||
+                isExtensionCompatibleQuery.isFetching
+              }
+              bg="backgroundHighlight"
+              w="full"
+              {...register("version", {
+                required: "Version is required",
+              })}
+            >
+              <option value="latest">Latest</option>
+              {allVersions?.data?.map(({ version }) => (
+                <option key={version} value={version}>
+                  {version}
+                </option>
+              ))}
+            </Select>
+          </Skeleton>
+          <FormErrorMessage>
+            {errors.version && errors.version.message}
+          </FormErrorMessage>
+        </FormControl>
+      </Grid>
+
+      <Spacer h={5} />
+
+      {/* Submit */}
+      <Flex justify={"flex-end"}>
+        <TransactionButton
+          transactionCount={1}
+          isLoading={installMutation.isLoading}
+          type="submit"
+          colorScheme="primary"
+          alignSelf="flex-end"
+          isDisabled={
+            !formState.isValid ||
+            isExtensionCompatibleQuery.data === false ||
+            isExtensionCompatibleQuery.isFetching
+          }
+        >
+          Install
+        </TransactionButton>
+      </Flex>
+    </form>
+  );
+};
+
+async function isExtensionCompatible(options: {
+  contractInfo: {
+    bytecode: string;
+    installedExtensionBytecodes: string[];
+    chainId: number;
+  };
+  extensionInfo: {
+    bytecodeUri: string;
+  };
+}) {
+  // 1. get extension's bytecode
+  const res = await download({
+    client: thirdwebClient,
+    uri: options.extensionInfo.bytecodeUri,
+  });
+
+  const extensionBytecode = await res.text();
+
+  // 2. check compatibility with core and installed extensions
+  try {
+    const isCompatible = await compatibleExtensions(
+      options.contractInfo.bytecode,
+      [...options.contractInfo.installedExtensionBytecodes, extensionBytecode],
+      options.contractInfo.chainId,
+    );
+
+    return isCompatible;
+  } catch (e) {
+    console.error("Error during compatibility check:", e);
+    // assume it's compatible - it's better to let the user install the extension - it will fail if it's not compatible anyway
+    throw e;
```

Comment:
the comment and the action (re-throwing the error) does not make sense to me, if we want to install the extension anyways shouldn't we return `true` here?

Reply:
comment is outdated - it used to return true, but does not anymore. removing comment